work/worker.go

```diff
@@ -186,9 +184,6 @@ func newWorker(config *params.ChainConfig, engine consensus.Engine, rewardbase c
 	go worker.update()
 
 	go worker.wait(TxResendUseLegacy)
-	worker.commitNewWork()
```

Comment:
This line is removed not to call commitNewWork before calling`woker.start()`.
In that case, there is nil `chain`. It causes `sb.GetProposer` access nil `chain` to get last block. 

`worker.commitNewWork()` is called by many cases like below. So we can remove this.
![image](https://user-images.githubusercontent.com/29390878/107925487-dcdb7400-6fb7-11eb-9b0f-5f759cc4c04e.png)

Reply:
I understood that `sb.chain` can be `nil`, however, why it wasn't a problem before?

Reply:
There was not a code (like `Prepare`) to use chain before setting `sb.chain`.
So It was not a problem. But calling `worker.commitNewWork()`  before `worker.start()` is weird.